messaging/connectors/kafka/src/main/java/io/helidon/messaging/connectors/kafka/KafkaConnector.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.messaging.connectors.kafka;
+
+import java.io.Closeable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.BeforeDestroyed;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import io.helidon.common.configurable.ScheduledThreadPoolSupplier;
+import io.helidon.config.Config;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.eclipse.microprofile.reactive.messaging.spi.IncomingConnectorFactory;
+import org.eclipse.microprofile.reactive.messaging.spi.OutgoingConnectorFactory;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+
+/**
+ * Implementation of Kafka Connector as described in the MicroProfile Reactive Messaging Specification.
+ */
+@ApplicationScoped
+@Connector(KafkaConnector.CONNECTOR_NAME)
+public class KafkaConnector implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    /**
+     * Microprofile messaging Kafka connector name.
+     */
+    static final String CONNECTOR_NAME = "helidon-kafka";
+    private static final Logger LOGGER = Logger.getLogger(KafkaConnector.class.getName());
+    private final ScheduledExecutorService scheduler;
+    private final Queue<Closeable> resourcesToClose = new LinkedList<>();
+
+    /**
+     * Constructor to instance KafkaConnectorFactory.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     */
+    @Inject
+    public KafkaConnector(Config config) {
```

Comment:
If this class should be part of Helidon SE, then the public constructor is an issue.
Also you need to add a Builder for any configurable options - in Helidon SE, most of things that can be done using `Config` should be doable using `Builder` and vice-versa

Reply:
For correct SE/MP split, I think you should:

1. Remove public constructor from this class
2. Create a builder that also supports config through `config(Config)` method as other builders in Helidon
3. Add a CDI extension for Kafka connector to another module, that would create the connector instance for MP messaging correctly injecting values using CDI